Added ability to add to Spark's Hadoop configuration

hntd187 opened the pull request:
I wanted to be able to add things to the Spark Hadoop configuration, specifically things like the shuffle manager and ParquetOutputCommitter which require you to set them via the hadoop configuration.

I added the ability to be able to add to the spark context's hadoop configuration. I think I may have to add this to the extras-api stuff too.

velvia commented on job-server/src/spark.jobserver/JobManagerActor.scala:
```diff
@@ -152,7 +158,23 @@ class JobManagerActor(contextConfig: Config) extends InstrumentedActor {
+    case ContextConfig => {
+      if (jobContext.sparkContext == null){
+        sender ! SparkContextDead
+      }else{
```
(nit) space before and after else

velvia commented on job-server/src/spark.jobserver/JobManagerActor.scala:
```diff
@@ -152,7 +158,23 @@ class JobManagerActor(contextConfig: Config) extends InstrumentedActor {
+      }
+    }
+    case ContextConfig => {
+      if (jobContext.sparkContext == null){
```
(nit) space before {

velvia commented on job-server/src/spark.jobserver/JobManagerActor.scala:
```diff
@@ -20,9 +24,11 @@ object JobManagerActor {
   case class StartJob(appName: String, classPath: String, config: Config,
                       subscribedEvents: Set[Class[_]])
   case class KillJob(jobId: String)
+  case object ContextConfig
```
It's slightly confusing to have the return message and the sending message be the same name.  How about rename this to `GetContextConfig`?

velvia commented:
@hntd187 many thanks!   This looks good overall, just some minor spacing and naming comments.  Also would you mind updating two more things:
1) README, mentioning this ability to set Hadoop config, and
2) application.conf, add a (perhaps commented out) section for the hadoop conf so users know where to add settings.

I have a big change to the API coming up.  Might borrow some of this going forward, so I hope this gets merged soon  :)

hntd187 commented:
@velvia awesome I'll make your changes. Also, you're gonna be giving a talk at ETE in philly this coming week right? I'll be there and I wanted to know if I could grab a few minutes of your time. Namely, I made some large changes in a branch for SJS for my job namely I upgraded a lot of the dependencies and I rewrote a lot of the metrics to use the latest version of dropwizard from the yammer. I figured submitting a pull request with such a huge amount of changes without talking to you first would be a bad idea. I also wanted to upgrade the slick install to the latest version but I hadn't gotten around to it. I also added a good bit of new routes for vision into how a job is currently executing and the current context configure and such. I'd ideally like to give all this back but I think my formatting and everything is all over the place with it so I thought I'd try and catch you for a few at ETE to talk about how to give it back. I have no issues going back and splicing the code up into small commits to submit. But since you have a big change too maybe my changes aren't necessary.

hntd187 commented:
Oh but with all those changes all your tests still pass unaltered so that's nice.

velvia commented:
Yeah definitely.  Let's meet up. Want to ping me on email/gchat/etc and we can set up a time?

-Evan
"Never doubt that a small group of thoughtful, committed citizens can change the world" - M. Mead

> On Apr 8, 2016, at 4:11 AM, Stephen Carman <notifications@github.com> wrote:
> 
> @velvia awesome I'll make your changes. Also, you're gonna be giving a talk at ETE in philly this coming week right? I'll be there and I wanted to know if I could grab a few minutes of your time. Namely, I made some large changes in a branch for SJS for my job namely I upgraded a lot of the dependencies and I rewrote a lot of the metrics to use the latest version of dropwizard from the yammer. I figured submitting a pull request with such a huge amount of changes without talking to you first would be a bad idea. I also wanted to upgrade the slick install to the latest version but I hadn't gotten around to it. I also added a good bit of new routes for vision into how a job is currently executing and the current context configure and such. I'd ideally like to give all this back but I think my formatting and everything is all over the place with it so I thought I'd try and catch you for a few at ETE to talk about how to giv  e it back. I have no issues going back and splicing the code up into small commits to submit. But since you have a big change too maybe my changes aren't necessary.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>

velvia commented:
@hntd187 many thanks, going to merge.
